src/ngraph/runtime/cpu/pass/cpu_collapse_dims.cpp

```diff
@@ -244,29 +245,29 @@ bool runtime::cpu::pass::CPUCollapseDims::run_on_function(std::shared_ptr<ngraph
     bool replaced = false;
     for (auto n : f->get_ordered_ops())
     {
-        if (std::dynamic_pointer_cast<op::Broadcast>(n))
+        if (as<OP_TYPEID::Broadcast>(n))
```

Comment:
Given that the implementation of `as<OP_TYPEID::SomeOpType>(n))` uses `get_typeid(n->description())`, which returns an `OP_TYPEID` that could be used in a switch statement, why not simplify this?
Instead of the templatized `as` trick, why not provide a helper method `get_typeid` which takes an `Op` and produces the `OP_TYPEID`. Then switch off that?